database/src/main/postgres/flows/V1.7.5__get_flow_checkpoints.sql

```diff
@@ -0,0 +1,127 @@
+    OUT checkpoint_end_time   TIMESTAMP WITH TIME ZONE
+) RETURNS SETOF record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: flows.get_flow_checkpoints(3)
+--      Retrieves all checkpoints (measures and their measurement details) related to a primary flow
+--      associated with the input partitioning.
+--
+-- Note: a single row returned from this function doesn't contain all data related to a single checkpoint - it only
+--     represents one measure associated with a checkpoint. So even if only a single checkpoint would be retrieved,
+--     this function can potentially return multiple rows.
+--
+-- Note: checkpoints will be retrieved in ordered fashion, by checkpoint_time and id_checkpoint
+--
+-- Parameters:
+--      i_partitioning_of_flow  - partitioning to use for identifying the flow associate with checkpoints
+--                                that will be retrieved
+--      i_limit                 - (optional) maximum number of checkpoints retrieved
+--                                if zero or lower specified, all data will be returned, i.e. no limit will be applied
+--      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
+--
+--      Note: checkpoint name uniqueness is not enforced by the data model, so there can be multiple different
+--          checkpoints with the same name in the DB, i.e. multiple checkpoints can be retrieved even when
+--          specifying `i_checkpoint_name` parameter
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_checkpoint       - id of retrieved checkpoint
+--      checkpoint_name     - name of retrieved checkpoint
+--      measure_name        - measure name associated with a given checkpoint
+--      measure_columns     - measure columns associated with a given checkpoint
+--      measurement_value   - measurement details associated with a given checkpoint
+--      checkpoint_time     - time
+--
+-- Status codes:
+--      11                  - Flow created
+--      16                  - No checkpoints found
+--      41                  - Partitioning not found
+--
+--
+-------------------------------------------------------------------------------
+
+DECLARE
+    _fk_partitioning BIGINT;
+BEGIN
+    _fk_partitioning = runs._get_id_partitioning(i_partitioning_of_flow);
+
+    IF _fk_partitioning IS NULL THEN
+        status := 41;
+        status_text := 'The partitioning does not exist.';
+        RETURN NEXT;
+        RETURN;
+    END IF;
+
+    RETURN QUERY
+    SELECT 11 AS status, 'OK' AS status_text,
+           CP.id_checkpoint, CP.checkpoint_name,
+           MD.measure_name, MD.measured_columns,
+           M.measurement_value,
+           CP.process_start_time AS checkpoint_start_time, CP.process_end_time AS checkpoint_end_time
+    FROM runs.checkpoints AS CP
+    JOIN runs.measurements AS M
```

Comment:
**Small:**
Would make it explicit it's an `INNER JOIN`

Reply:
every join (unless otherwise specified, e.g. `LEFT` or `RIGHT`) is `INNER` - I don't really prefer such 'waste' of space, everybody knows this in SQL world and thus in this case I don't like to be explicit :) 

(really, based on my experience almost no one these days writes INNER)

Do you insist on it though?

Reply:
Don't insist. But our bubbles have not intersection it seems. All people I remember wrote SQL write `INNER JOIN`. 😄 
Even in the code I saw yesterday in Absa... (will send in priv)